### PR TITLE
Expose Tendermint HTTP RPC

### LINKED ISSUE #172: RPC configuration is hardcoded
In "local RPC" implementation, configuration is [hardcoded](https://github.com/celestiaorg/optimint/blob/39ffb65790ea320217708a5eff734568b4dffceb/rpcclient/local.go#L43).

~~Because of that, starport generated applications are failing not fully functional.~~

### LINKED ISSUE #173: Expose Tendermint RPC
Optimint should expose Tendermint RPC for full starport compatibility.

### PR DESCRIPTION
**Pull request by tzdybal**
Resolves #172.
Resolves #173.

### CODE REVIEW

**Comment by tzdybal** [comment]
I marked this PR as ready for review. I'll prepare another PR for WebSocket support.

**Comment by tzdybal** [comment]
The current implementation is able to actually process JSON-RPC requests, including the ones from starport (with integers quoted as strings).

**Comment by tzdybal** [comment]
Many methods are not implemented yet, so they do cause (expected) panics!

**Review by liamsi — APPROVED** [review]

File: conv/abci/block.go [hunk 1]
```diff
@@ -38,3 +38,49 @@ func ToABCIHeader(header *types.Header) (tmproto.Header, error) {
...
+		commitSig := tmproto.CommitSig{
+			BlockIdFlag:      tmproto.BlockIDFlagCommit,
+			//ValidatorAddress: nil,
+			//Timestamp:        nil,
+			Signature:        sig,
+		}
+		tmCommit.Signatures = append(tmCommit.Signatures, commitSig)
+	}
+	return &tmCommit
+}
```
Comment (adlerjohn) [inline conv/abci/block.go:86]: I notice some no-newlines here. I don't think CI is running `gofmt`. Can that be added?

File: conv/abci/block.go [hunk 2]
```diff
@@ -38,3 +38,49 @@ func ToABCIHeader(header *types.Header) (tmproto.Header, error) {
...
+		BlockID:    tmproto.BlockID{
+			Hash:          commit.HeaderHash[:],
+			PartSetHeader: tmproto.PartSetHeader{},
+		},
+	}
+	for _, sig := range commit.Signatures {
+		commitSig := tmproto.CommitSig{
+			BlockIdFlag:      tmproto.BlockIDFlagCommit,
+			//ValidatorAddress: nil,
+			//Timestamp:        nil,
```
Comment (adlerjohn) [inline conv/abci/block.go:80]: What are the ramifications of commenting this out vs explicitly setting `nil`?

File: rpc/json/service.go [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+	client  *client.Client
+	methods map[string]*method
+}
+
+func newService(c *client.Client) *service {
+	s := service{
+		client: c,
+	}
+	s.methods = map[string]*method{
+		"subscribe":            newMethod(s.Subscribe),
```
Comment (adlerjohn) [inline rpc/json/service.go:46]: Are these from somewhere? If not, can they be sorted alphabetically?

File: conv/abci/block.go (hunk 2 above)
Comment (tzdybal) [reply conv/abci/block.go:80]: you can't set nil, as its `string` and `time.Time` ;)
TBH it's a dirty work-in-progress version, and I'm reworking this part anyways. I'll move changes to separate branch/PR.

File: rpc/json/service.go (hunk 3 above)
Comment (tzdybal) [reply rpc/json/service.go:46]: It's all copy+paste+regexp from [tendermint](https://github.com/tendermint/tendermint/blob/master/internal/rpc/core/routes.go).

File: conv/abci/block.go (hunk 2 above)
Comment (tzdybal) [reply conv/abci/block.go]: Changes moved to #187.

**Pull request merged**